### PR TITLE
fixes for running with docker-compose

### PR DESCRIPTION
**Pull request by hellonico**
Added missing software, and files to run in docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ FROM alpine:edge
 
 RUN apk add --no-cache \
     # Runtime deps
-    ruby ruby-bundler ruby-bigdecimal ruby-io-console tzdata nodejs bash \
+    ruby ruby-bundler ruby-bigdecimal ruby-io-console tzdata nodejs bash git \
```
Comment (glebm) [inline Dockerfile]: What do we need git for?

File: docker-compose.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ web:
     DB_PASSWORD:
   command: >
     bash -c 'script/wait-for-tcp $$DB_1_PORT_5432_TCP_ADDR $$DB_1_PORT_5432_TCP_PORT &&
+             bundle install &&
```
Comment (glebm) [inline docker-compose.yml]: `bundle --path=$BUNDLE_PATH` already runs via Dockerfile:24

File: Dockerfile (hunk 1 above)
Comment (hellonico) [reply Dockerfile]: could not get bundler to install the git-based gems without it ...

File: docker-compose.yml (hunk 2 above)
Comment (hellonico) [reply docker-compose.yml]: dependencies were not met at runtime (docker-compose up)  without it.

File: Dockerfile (hunk 1 above)
Comment (glebm) [reply Dockerfile]: Got it, thanks. Please move to the line below "Bundle install deps"

File: docker-compose.yml (hunk 2 above)
Comment (glebm) [reply docker-compose.yml]: Perhaps this happened because your other fixes were not applied yet?
This really shouldn't be necessary.

Comment (hellonico) [reply docker-compose.yml]: I know ... trying one more time now without that extra line.

File: Dockerfile (hunk 1 above)
Comment (hellonico) [reply Dockerfile]: For info, list of the git based gems

```
Fetching https://github.com/rspec/rspec-rails
Fetching https://github.com/rspec/rspec-core
Fetching https://github.com/rspec/rspec-mocks
Fetching https://github.com/rspec/rspec-support
Fetching https://github.com/rspec/rspec-expectations
```

Comment (glebm) [reply Dockerfile]: I've removed them now (https://github.com/thredded/thredded/commit/38946f2cef65b93f7df59b3f7bdbedc292283ff2) but still let's keep `git` here as we occasionally need git gems, especially when adding support for new Rails versions.

File: docker-compose.yml (hunk 2 above)
Comment (hellonico) [reply docker-compose.yml]: When doing docker-compose up...

```
web_1  | bundler: failed to load command: rake (/bundle/ruby/2.5.0/bin/rake)
web_1  | Bundler::GitError: The git source https://github.com/rspec/rspec-rails is not yet checked out. Please run `bundle install` before trying to start your application
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git/git_proxy.rb:235:in `allowed_in_path'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git/git_proxy.rb:192:in `find_local_revision'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git/git_proxy.rb:64:in `revision'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git.rb:225:in `revision'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git.rb:93:in `install_path'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/path.rb:126:in `expanded_path'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/path.rb:163:in `load_spec_files'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git.rb:200:in `load_spec_files'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/path.rb:100:in `local_specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/source/git.rb:167:in `specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:758:in `block in converge_locked_specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:744:in `each'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:744:in `converge_locked_specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:247:in `resolve'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:170:in `specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:237:in `specs_for'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/definition.rb:226:in `requested_specs'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/runtime.rb:108:in `block in definition_method'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/runtime.rb:20:in `setup'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler.rb:107:in `setup'
web_1  |   /usr/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/setup.rb:20:in `<top (required)>'
web_1  |   /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
web_1  |   /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
thredded_web_1 exited with code 1

```

Comment (glebm) [reply docker-compose.yml]: I'm getting a different error:

```
web_1  | /usr/lib/ruby/2.5.0/rubygems.rb:284:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /thredded/Gemfile.lock. (Gem::GemNotFoundException)
```

Look into it

Comment (glebm) [reply docker-compose.yml]: Ah, looks like the issue is due to a conflict between the local Gemfile.lock and the one in the Docker container (they are the same file due to how the volumes are mounted)

**Comment by glebm** [comment]
Fixed in #816

**Comment by glebm** [comment]
More fixes in https://github.com/thredded/thredded/commit/7353061696d160315db35e9a1e4b5e747edfe7a0

You may have to recreate the existing containers:

```bash
docker-compose down --rmi all
docker-compose up
```

**Comment by glebm** [comment]
Pushed a few more fixes, should be fully working now

**Pull request closed without merging**